src/util/report.py

```diff
@@ -126,6 +126,79 @@ def add_to_migration_reports(message, category="Other", format="text"):
         _logger.warning("Upgrade report is growing suspiciously long: %s characters so far.", migration_reports_length)
 
 
+report = add_to_migration_reports
+
+
+def report_with_summary(summary, details, category="Other"):
+    """Append the upgrade report with a new entry.
+
+    :param str summary: Description of a report entry.
+    :param str details: Detailed description that is going to be folded by default.
+    :param str category: Title of a report entry.
+    """
+    msg = (
+        "<summary>{}<details>{}</details></summary>".format(summary, details)
+        if details
+        else "<summary>{}</summary>".format(summary)
+    )
+    report(message=msg, category=category, format="html")
+
+
+def report_with_list(summary, data, columns, row_format, links=None, total=None, limit=100, category="Other"):
+    """Append the upgrade report with a new entry that displays a list of records.
+
+    The entry consists of a category (title) and a summary (body).
+    The entry displays a list of records previously returned by an SQL query, or any list as long as it's passed as a Python List of Tuples.
+
+    .. example::
+
+        .. code-block:: python
+
+            util.report_with_list(summary="The following records were altered.",
+                                  data=cr.fetchall(),
+                                  columns=("id", "name", "city", "comment", "company_id", "company_name")
+                                  row_format="Partner with id {partner_link} works at company {company_link} in {city}, ({comment})",
+                                  links={"company_link": ("res.company", "company_id", "company_name"), "partner_link": ("res.partner", "id", "name")},
+                                  category="Accounting"
+                                  )
```

Comment:
The usual formatting allows for shorter lines. Unless there is some rendering issue?
```suggestion
            util.report_with_list(
                summary="The following records were altered.",
                data=cr.fetchall(),
                columns=("id", "name", "city", "comment", "company_id", "company_name")
                row_format="Partner with id {partner_link} works at company {company_link} in {city}, ({comment})",
                links={"company_link": ("res.company", "company_id", "company_name"), "partner_link": ("res.partner", "id", "name")},
                category="Accounting"
            )
```